### PR TITLE
fix: create month-end cron backup error

### PR DESCRIPTION
**Pull request by Metrora**
Signed-off-by: Liucw <liu.chuwei@99cloud.net>

### What type of PR is this?
/kind bug

### Which issue(s) this PR fixes:

Fixes https://github.com/kubeclipper-labs/kubeclipper/issues/110

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```
@zhuzhenfan @x893675

### CODE REVIEW

**Review by zhuzhenfan — APPROVED** [review]

File: pkg/apis/core/v1/handler.go [hunk 1]
```diff
@@ -2754,11 +2754,14 @@ func (h *handler) CreateCronBackup(request *restful.Request, response *restful.R
 		return
 	}
 	if cb.Spec.Schedule != "" {
-		s, _ := cron.NewParser(4 | 8 | 16 | 32 | 64).Parse(cb.Spec.Schedule)
+		schedule := ParseSchedule(cb.Spec.Schedule)
```
Comment (x893675) [inline pkg/apis/core/v1/handler.go:2757]: move this to controller, api handler only check schedule is valid

File: pkg/apis/core/v1/utils.go [hunk 2]
```diff
@@ -384,3 +386,22 @@ func (h *handler) checkBackupPointInUse(backups *v1.BackupList, name string) boo
 	}
 	return false
 }
+
+func ParseSchedule(schedule string) string {
```
Comment (x893675) [inline pkg/apis/core/v1/utils.go:390]: add unit test

File: pkg/controller/cronbackupcontroller/controller.go [hunk 3]
```diff
@@ -117,6 +126,18 @@ func (r *CronBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if cronBackup.Spec.Schedule != "" {
+		// first created cron backup, update next schedule time
+		if cronBackup.Status.NextScheduleTime == nil {
+			schedule := v1core.ParseSchedule(cronBackup.Spec.Schedule)
```
Comment (x893675) [inline pkg/controller/cronbackupcontroller/controller.go:131]: move function to controller package

File: pkg/controller/cronbackupcontroller/controller.go [hunk 4]
```diff
@@ -117,6 +126,18 @@ func (r *CronBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request)
...
+		if cronBackup.Status.NextScheduleTime == nil {
+			schedule := parseSchedule(cronBackup.Spec.Schedule)
+			s, _ := cron.NewParser(4 | 8 | 16 | 32 | 64).Parse(schedule)
+			// update the next schedule time
+			nextRunAt := metav1.NewTime(s.Next(time.Now()))
+			cronBackup.Status.NextScheduleTime = &nextRunAt
+			_, err = r.CronBackupWriter.UpdateCronBackup(ctx, cronBackup)
+			if err != nil {
+				log.Error("Failed to update cronBackup", zap.Error(err))
+			}
```
Comment (x893675) [inline pkg/controller/cronbackupcontroller/controller.go:139]: return ctrl.Result{}, nil

File: pkg/controller/cronbackupcontroller/controller.go [hunk 5]
```diff
@@ -93,6 +94,14 @@ func (r *CronBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	now := metav1.NewTime(time.Now())
 	if cronBackup.Spec.RunAt != nil {
+		// first created cron backup, update next schedule time
+		if cronBackup.Status.NextScheduleTime == nil {
+			cronBackup.Status.NextScheduleTime = cronBackup.Spec.RunAt
+			_, err = r.CronBackupWriter.UpdateCronBackup(ctx, cronBackup)
+			if err != nil {
+				log.Error("Failed to update cronBackup", zap.Error(err))
+			}
```
Comment (x893675) [inline pkg/controller/cronbackupcontroller/controller.go:103]: ctrl.Result{}, nil

**Pull request merged**